src/server.c

```diff
@@ -5493,32 +5496,19 @@ int processCommand(client *c) {
         return C_OK;
     }
 
-    /* Lua script too slow? Only allow a limited number of commands.
+    /* when a busy job is being done (Lua script/Module command)
+     * Only allow a limited number of commands.
      * Note that we need to allow the transactions commands, otherwise clients
      * sending a transaction with pipelining without error checking, may have
      * the MULTI plus a few initial commands refused, then the timeout
      * condition resolves, and the bottom-half of the transaction gets
      * executed, see Github PR #7022. */
-    if (scriptIsTimedout() &&
-          c->cmd->proc != authCommand &&
-          c->cmd->proc != helloCommand &&
-          c->cmd->proc != replconfCommand &&
-          c->cmd->proc != multiCommand &&
-          c->cmd->proc != discardCommand &&
-          c->cmd->proc != watchCommand &&
-          c->cmd->proc != unwatchCommand &&
-          c->cmd->proc != quitCommand &&
-          c->cmd->proc != resetCommand &&
-        !(c->cmd->proc == shutdownCommand &&
-          c->argc == 2 &&
-          tolower(((char*)c->argv[1]->ptr)[0]) == 'n') &&
-        !(c->cmd->proc == scriptCommand &&
-          c->argc == 2 &&
-          tolower(((char*)c->argv[1]->ptr)[0]) == 'k') &&
-        !(c->cmd->proc == functionsKillCommand) &&
-        !(c->cmd->proc == functionsStatsCommand))
-    {
-        if (scriptIsEval()) {
+
+    if (c->cmd->proc == shutdownCommand && c->argc == 2 &&
+      tolower(((char*)c->argv[1]->ptr)[0]) == 's') c->cmd->flags |= ~CMD_ALLOW_BUSY;
```

Comment:
First, this is very wrong (modifying the command flags, permanently), as soon as it is called with some argument.

but anyway, you'll soon be able to revert this change, since this is handled properly in #9872 (i.e. it allows the entire SHUTDOWN command, and the command itself does the rest of the filtering)